src/BaseFactory.php

```diff
@@ -33,7 +35,16 @@ public static function new(): self
 
     protected function build(array $extra = [], string $creationType = 'create')
     {
-        $model = $this->modelClass::$creationType(array_merge($this->getDefaults($this->faker), $this->overwriteDefaults, $extra));
+        $modelData = collect(array_merge(
```

Comment:
Can you create a new private method for that so that we can write here

```php
$model = $this->modelClass::$creationType($this->prepareModelDataForBuild());
```